apollo-adapters-core/src/commonMain/kotlin/com/apollographql/adapter/core/UnitAdapter.kt

```diff
@@ -0,0 +1,16 @@
+package com.apollographql.adapter.core
+
+import com.apollographql.apollo.api.Adapter
+import com.apollographql.apollo.api.CustomScalarAdapters
+import com.apollographql.apollo.api.json.JsonReader
+import com.apollographql.apollo.api.json.JsonWriter
+
+internal object UnitAdapter : Adapter<Unit> {
+  override fun fromJson(reader: JsonReader, customScalarAdapters: CustomScalarAdapters): Unit {
+    return Unit
```

Comment:
We should probably call `reader.skipValue()` here. The reason it works in our case is that the backend always return `null` for `Void` (and so `NullableAdapter` doesn't delegate), but it may be different for other backends.

Reply:
Done :+1: